### PR TITLE
Feature/#157 Configured pipeline for state management in Gitlab

### PR DESCRIPTION
**Pull request by jomcy-amzn**
Added pre-commit config.yaml
Removed cluster group in gitlab

*Issue #157, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Zvikan** [comment]
CC @naris-silpakit

File: examples/ci_cd/gitlab-ci-cd/main.tf [hunk 1]
```diff
@@ -61,29 +61,20 @@ data "aws_eks_cluster_auth" "cluster" {
...
-  terraform_version = "Terraform v1.0.1"
+  vpc_name     = join("-", [var.tenant, var.environment, var.zone, "vpc"])
+  cluster_name = join("-", [var.tenant, var.environment, var.zone, "eks"])
 }
 
 module "aws_vpc" {
-  source  = "terraform-aws-modules/vpc/aws"
-  version = "v3.2.0"
+  source = "terraform-aws-modules/vpc/aws"
+  //version = "v3.2.0"
```
Comment (naris-silpakit) [inline examples/ci_cd/gitlab-ci-cd/main.tf:70]: Can we uncomment this? I accidentally removed it before, but I understand now it's used for recording the module version.

File: examples/ci_cd/gitlab-ci-cd/README.md [hunk 2]
```diff
@@ -1,30 +1,61 @@
...
+    - Make sure the GitLab group that you're referencing in `data.gitlab_group.gitops-eks` exists and that you have access to it.
+
+- ### 400 Error when creating resource
+
+    - If the error contains `{message: {environment_scope: [cannot add duplicated environment scope]}}`, it is likely that an existing Kubernetes integration with the same environment scope was not removed. Remove any Kubernetes clusters with the same environment scope from the GitLab group before redeploying.
 
-## Step5: Commit changes to verify the pipeline
+- ### What's gitlab-terraform?
 
+    - `gitlab-terraform` is a thin wrapper around the `terraform` binary as part of the [GitLab Terraform docker image](registry.gitlab.com/gitlab-org/terraform-images/stable:latest) used in `.gitlab-ci.yml`.
```
Comment (naris-silpakit) [inline examples/ci_cd/gitlab-ci-cd/README.md:56]: Can you update the link from registry.gitlab.com/gitlab-org/terraform-images/stable:latest to https://gitlab.com/gitlab-org/terraform-images? For context: https://github.com/aws-samples/aws-eks-accelerator-for-terraform/pull/212#discussion_r790006559

File: examples/ci_cd/gitlab-ci-cd/main.tf [hunk 3]
```diff
@@ -61,29 +61,20 @@ data "aws_eks_cluster_auth" "cluster" {
...
 
   name = local.vpc_name
-  cidr = local.vpc_cidr
+  cidr = var.vpc_cidr
   azs  = data.aws_availability_zones.available.names
 
-  public_subnets  = [for k, v in slice(data.aws_availability_zones.available.names, 0, 3) : cidrsubnet(local.vpc_cidr, 8, k)]
-  private_subnets = [for k, v in slice(data.aws_availability_zones.available.names, 0, 3) : cidrsubnet(local.vpc_cidr, 8, k + 10)]
+  public_subnets  = [for k, v in data.aws_availability_zones.available.names : cidrsubnet(var.vpc_cidr, 8, k)]
+  private_subnets = [for k, v in data.aws_availability_zones.available.names : cidrsubnet(var.vpc_cidr, 8, k + 10)]
```
Comment (naris-silpakit) [inline examples/ci_cd/gitlab-ci-cd/main.tf:77]: Can you add the slice back? I removed it erroneously: https://github.com/aws-samples/aws-eks-accelerator-for-terraform/pull/212#discussion_r790018967

File: examples/ci_cd/gitlab-ci-cd/.pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      - id: end-of-file-fixer
+  - repo: https://github.com/antonbabenko/pre-commit-terraform
+    rev: v1.62.3
+    hooks:
+      - id: terraform_fmt
+      - id: terraform_validate
+      - id: terraform_tflint
+      - id: terraform_tfsec
+      - id: terraform_docs
+default_stages: [commit, push]
```
Comment (vara-bonthu) [inline examples/ci_cd/gitlab-ci-cd/.pre-commit-config.yaml:23]: Is there any reason why we are adding this to examples? Please remove this file

File: examples/ci_cd/gitlab-ci-cd/input.tfvars [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# gitlab-provider.tf
+group_id          = 19364 //Refer GitlabUI to find your group ID
+gitlab_project_id = 20206 ///Refer GitlabUI to find your Project ID
+
+# main.tf
+tenant             = "aws002"
+environment        = "preprod"
+zone               = "jimqa"
```
Comment (vara-bonthu) [inline examples/ci_cd/gitlab-ci-cd/input.tfvars:8]: change this to dev or test

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
Looks Good! Few comments needs addressing

File: examples/ci_cd/gitlab-ci-cd/main.tf (hunk 1 above)
Comment (jomcy-amzn) [reply examples/ci_cd/gitlab-ci-cd/main.tf:70]: Done! updated

File: examples/ci_cd/gitlab-ci-cd/main.tf (hunk 3 above)
Comment (jomcy-amzn) [reply examples/ci_cd/gitlab-ci-cd/main.tf:77]: Done , Updated the file

File: examples/ci_cd/gitlab-ci-cd/main.tf (hunk 1 above)
Comment (vara-bonthu) [inline examples/ci_cd/gitlab-ci-cd/main.tf]: version needs to be uncommented

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
There are still few comments outstanding that needs addressing

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Review by askulkarni2 — APPROVED** [review]
LGTM

**Pull request merged**